src/System.Drawing.Primitives/src/System/Drawing/PointF.cs

```diff
@@ -0,0 +1,205 @@
+        {
+            return Subtract(pt, sz);
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Compares two <see cref='System.Drawing.PointF'/> objects. The result specifies
+        ///       whether the values of the <see cref='System.Drawing.PointF.X'/> and <see cref='System.Drawing.PointF.Y'/> properties of the two <see cref='System.Drawing.PointF'/>
+        ///       objects are equal.
+        ///    </para>
+        /// </summary>
+        public static bool operator ==(PointF left, PointF right)
+        {
+            return left.X == right.X && left.Y == right.Y;
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Compares two <see cref='System.Drawing.PointF'/> objects. The result specifies whether the values
+        ///       of the <see cref='System.Drawing.PointF.X'/> or <see cref='System.Drawing.PointF.Y'/> properties of the two
+        ///    <see cref='System.Drawing.PointF'/> 
+        ///    objects are unequal.
+        /// </para>
+        /// </summary>
+        public static bool operator !=(PointF left, PointF right)
+        {
+            return !(left == right);
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Translates a <see cref='System.Drawing.PointF'/> by a given <see cref='System.Drawing.Size'/> .
+        ///    </para>
+        /// </summary>
+        public static PointF Add(PointF pt, Size sz)
+        {
+            return new PointF(pt.X + sz.Width, pt.Y + sz.Height);
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Translates a <see cref='System.Drawing.PointF'/> by the negative of a given <see cref='System.Drawing.Size'/> .
+        ///    </para>
+        /// </summary>
+        public static PointF Subtract(PointF pt, Size sz)
+        {
+            return new PointF(pt.X - sz.Width, pt.Y - sz.Height);
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Translates a <see cref='System.Drawing.PointF'/> by a given <see cref='System.Drawing.SizeF'/> .
+        ///    </para>
+        /// </summary>
+        public static PointF Add(PointF pt, SizeF sz)
+        {
+            return new PointF(pt.X + sz.Width, pt.Y + sz.Height);
+        }
+
+        /// <summary>
+        ///    <para>
+        ///       Translates a <see cref='System.Drawing.PointF'/> by the negative of a given <see cref='System.Drawing.SizeF'/> .
+        ///    </para>
+        /// </summary>
+        public static PointF Subtract(PointF pt, SizeF sz)
+        {
+            return new PointF(pt.X - sz.Width, pt.Y - sz.Height);
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (!(obj is PointF))
+                return false;
+
+            PointF comp = (PointF)obj;
+            return comp.X == X && comp.Y == Y;
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
```

Comment:
We should provide a better GetHashCode implementation (since this is a port of what's in the full framework, we could do that separately).  The base ValueType's GetHashCode is fairly expensive.